source_gen/lib/src/generator_for_annotation.dart

```diff
@@ -66,6 +66,6 @@ abstract class GeneratorForAnnotation<T> extends Generator {
   ///
   /// Implementations should return `null` when no content is generated. Empty
   /// or whitespace-only [String] instances are also ignored.
-  generateForAnnotatedElement(
+  dynamic generateForAnnotatedElement(
```

Comment:
Shouldn't this really be `FutureOr<String>` also?

Reply:
See the doc comment. This can return an iterable or Stream, too...